cockatrice/src/releasechannel.cpp

```diff
@@ -219,63 +221,75 @@ QString DevReleaseChannel::getReleaseChannelUrl() const
     return QString(DEVRELEASE_URL);
 }
 
+
 void DevReleaseChannel::releaseListFinished()
 {
     QNetworkReply *reply = static_cast<QNetworkReply *>(sender());
-    bool ok;
-    QString tmp = QString(reply->readAll());
+    QByteArray jsonData = reply->readAll();
     reply->deleteLater();
 
-    QVariantMap resultMap = QtJson::Json::parse(tmp, ok).toMap();
-    if (!ok) {
-        qWarning() << "No reply received from the release update server:" << tmp;
+    QJsonDocument doc = QJsonDocument::fromJson(jsonData);
+    QJsonArray array = doc.array();
+
+    /*
+     * Get the latest release on GitHub
+     * This can be either a pre-release or a published release
+     * depending on timing. Both are acceptable.
+     */
+    QVariantMap resultMap = array.at(0).toObject().toVariantMap();
```

Comment:
`array.at(0)` can fail if we ever deleted the releases or if the github api for some reason returned bad json.

Does the app crash in that case?